packages/gatsby-theme/src/components/Account.tsx

```diff
@@ -1,31 +1,29 @@
 import React from 'react';
-import { Query } from 'react-apollo';
-import { CurrentUserQuery } from '../graphqlTypes';
 import { Text, Image, Flex } from './shared/base';
 import { Link } from 'gatsby';
-import { CenteredLoader } from './Loader';
 import CustomButton from './CustomButton';
-import { optionalChaining } from '../utils/helpers';
-import { loginUser } from '../utils/auth';
-import { CURRENT_USER } from './queries/userQueries';
+import { loginUser } from '../utils/auth/auth';
+import WithCurrentUser from '../utils/auth/WithCurrentUser';
+import { CenteredLoader } from '../components/Loader';
 
 const Account = () => {
   return (
-    <Query<CurrentUserQuery> query={CURRENT_USER}>
-      {({ data, error, loading }) => {
-        if (error || loading) {
+    <WithCurrentUser>
+      {({ user, loading }) => {
+        if (loading) {
           return <CenteredLoader />;
         }
-        if (optionalChaining(() => data.viewer.user)) {
-          return <Profile user={data.viewer.user} />;
+        if (user) {
+          return <Profile user={user} />;
+        } else {
```

Comment:
```suggestion
        }
```

Reply:
Hey @rohit-ravikoti I'm not sure what this edit is for